### PR TITLE
[BD] Upgrade test requirements

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description 
Remove pined version from the file test.in Only the trivial upgrades.
this https://github.com/edx/edx-analytics-dashboard/pull/985/ is necessary. 
Part of .https://openedx.atlassian.net/browse/BOM-1360

### Reviewers
- [ ] @morenol 
- [x]  Is this ready for edX's review? 
 
### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @andrey-canon! I've created [OSPR-4331](https://openedx.atlassian.net/browse/OSPR-4331) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by morenol** [comment]
Hi @jmbowman, This is also ready for review.

File: requirements/test.in [hunk 1]
```diff
@@ -4,20 +4,20 @@
...
-coverage==4.4.1
-ddt==1.1.1
+coverage
+ddt
 django-dynamic-fixture
 elasticsearch==2.4.1
-unittest2==1.1.0
-httpretty==0.8.14
-mock==1.3.0
+unittest2
```
Comment (jmbowman) [inline requirements/test.in]: Is this actually still needed?  Looks like a backport of testing functionality already in Python 3.5+, don't see it imported anywhere in the code.

File: requirements/test.in [hunk 2]
```diff
@@ -4,20 +4,20 @@
...
+httpretty
+mock
 pep257
 pycodestyle
 pylint==1.7.1
-logilab-common==1.4.0
+logilab-common
```
Comment (jmbowman) [inline requirements/test.in]: This may be unused also.

File: requirements/test.in [hunk 3]
```diff
@@ -4,20 +4,20 @@
...
 selenium>=2.44.0,<3.0.0
 sure
```
Comment (jmbowman) [inline requirements/test.in]: Also unused?

File: requirements/test.in [hunk 4]
```diff
@@ -4,20 +4,20 @@
...
+coverage
+ddt
 django-dynamic-fixture
 elasticsearch==2.4.1
-unittest2==1.1.0
-httpretty==0.8.14
-mock==1.3.0
+unittest2
+httpretty
+mock
```
Comment (jmbowman) [inline requirements/test.in]: If the dependency on `sure` can be removed, using `unittest.mock` from the standard library instead would also get rid of this dependency.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**